sql-tuning-best-practice.md

```diff
@@ -0,0 +1,714 @@
+
+## Monitoring SQL Statements by Using TiDB Dashboard
+### SQL Statements Panel 
+In TiDB Dashboard, navigate to SQL Statements panel, which helps us identify the following:
+
+1. The SQL statement with the highest total latency is the one that takes the longest time to execute out of multiple executions of the same SQL statement. 
+2. It can also display the number of times each SQL statement has been executed cumulatively, allowing us to find the SQL statement with the highest execution frequency.
+3. Clicking on each SQL statement allows us to delve deeper into the EXPLAIN ANALYZE results.
+
+SQL statements are normalized as templates, where literals and bind variables are replaced by '?'. This normalization and sorting process allows you to quickly pinpoint the most resource-intensive queries that may require optimization.
+![sql-statements-default](/media/sql-tuning/sql-statements-default.png)
+
+
+### Slow Queries Panel Default Display
+In the TiDB Dashboard, we can find the Slow Query panel, which displays all SQL statements whose execution time exceeds the time threshold set by the system variable tidb_slow_log_threshold (default 300 milliseconds). 
+
+On Slow Queries panel, we can find:
+
+1. The slowest SQL queries.
+2. The SQL query that reads the most data from TiKV.
+3. The EXPLAIN ANALYZE output from drilling down the SQL statement by clicking it.
+4. Please note that on the Slow Queries panel, we cannot get the frequency of the SQL statement execution. Once the execution elapsed time exceeds tidb_slow_log_threshold for single instance, the query is then listed on the Slow Queries panel.
+![slow-query-default](/media/sql-tuning/slow-query-default.png)
+
+## Other Tools for Identifying Top SQL
+
+In addition to TiDB Dashboard, there are several other tools available to identify resource-intensive SQL queries:
+
+Each tool offers unique insights and can be valuable for different analysis scenarios. Using a combination of these tools allows for comprehensive SQL performance monitoring and optimization.
+- [slow query log](https://docs.pingcap.com/tidb/stable/identify-slow-queries)
+- [statements_summary view](https://docs.pingcap.com/tidb/stable/statement-summary-tables#statements_summary)
+- [top sql feature](https://docs.pingcap.com/tidb/stable/top-sql)
+- [expensive queries in tidb log](https://docs.pingcap.com/tidb/stable/identify-expensive-queries)
+- [cluster_processlist view](https://docs.pingcap.com/tidb/stable/information-schema-processlist#cluster_processlist)
+
+
+## Gathering Data on the SQL Identified
+
+For the top SQL statements identified, you can use [PLAN REPLAYER](https://docs.pingcap.com/tidb/stable/sql-plan-replayer) to capture and save the on-site information of a TiDB cluster. This tool allows you to recreate the execution environment for further analysis. The syntax for exporting SQL information is as follows:
```

Comment:
This and many other links seem to link with the full URL of the doc instead of directly to the right doc. This means that people might jump between TiDB versions.

```suggestion
For the top SQL statements identified, you can use [PLAN REPLAYER](/sql-plan-replayer.md) to capture and save the on-site information of a TiDB cluster. This tool allows you to recreate the execution environment for further analysis. The syntax for exporting SQL information is as follows:
```